src/main/java/pl/wedding/management/service/user/model/EMail.java

```diff
@@ -0,0 +1,45 @@
+package pl.wedding.management.service.user.model;
+
+import lombok.Getter;
+import org.springframework.util.StringUtils;
+
+import java.util.Objects;
+
+@Getter
+public class EMail {
+
+    private static final String AT_SIGN = "@";
+    private final String email;
+
+    public EMail(String email) {
+
+        if(isInvalid(email)){
+            throw new IllegalArgumentException();
+        }
+        this.email = email;
+    }
+
+    private boolean isInvalid(String value){
+
+        //TODO: Email Validation
+        if(StringUtils.isEmpty(value) || !value.contains(AT_SIGN)){
```

Comment:
You think that empty string is valid email address? Try sending mail there, and see what your mail client will tell you.

```suggestion
        return !StringUtils.isEmpty(value) && value.contains(AT_SIGN);
```

Email validation is hard to do. Specification is long, with many preceedors and security issues. You can use some open sourced software for handling some cases. It's not perfect, but **_decent_** solution, ex: [EmailValidator](https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/EmailValidator.html)